# Lecture 2b: Simulating a Multivariate Normal Data Matrix
**Simulations in Statistics (52001)**

---

## Overview

We construct a function `Part2b` that returns an $n \times d$ **data matrix** whose rows are independent copies of a multivariate Normal vector built by the linear transformation

$$\vec{X} = A\vec{Z} + \vec{b}, \qquad \vec{Z} \sim \mathcal{N}(\vec{0},\ I_k)$$

where $\vec{Z}$ is a vector of $k$ independent standard Normals.

**Arguments:**

- `n` -- integer, number of independent rows (observations).
- `b` -- length-$d$ vector, the additive component.
- `A` -- $d \times k$ matrix, the multiplicative component (rows of `A` $=$ length of `b`).

**Output:** matrix of dimension $n \times d$ where $d = $ `length(b)`.

**Constraint:** do **not** use `MASS::mvrnorm`. Build $X$ directly from `rnorm`.

**Distribution of each row:**

$$\vec{X}_i \sim \mathcal{N}_d(\vec{b},\ A A^\top)$$

so $\mathbb{E}[X_{ij}] = b_j$ and $\operatorname{Cov}(X_{ij}, X_{ik}) = (A A^\top)_{jk}$.

**Application:** $n = 10^6$, $\vec{b} = (0.67,\ -0.43,\ -0.17,\ 1.13,\ 0.29)$ and $A$ a $5 \times 3$ matrix (given below). Report:

$$\text{(a)}\;\; \operatorname{colMeans}(X)[1] \qquad \text{(b)}\;\; \operatorname{var}(X)[1,1] \qquad \text{(c)}\;\; \operatorname{var}(X)[1,2]$$

Theoretical targets: $b_1$, $(AA^\top)_{1,1}$, $(AA^\top)_{1,2}$.

## R Implementation

In [4]:
Part2b <- function(n, b, A) {
  d            <- length(b)
  k            <- ncol(A)
  Z_matrix     <- matrix(rnorm(n * k), nrow = k, ncol = n)
  AZ_plus_b    <- A %*% Z_matrix + as.vector(b)
  data_matrix  <- t(AZ_plus_b)
  return(data_matrix)
}

In [5]:
set.seed(1)

n_rows  <- 10^6
b_shift <- c(0.67, -0.43, -0.17, 1.13, 0.29)
A_mix   <- matrix(c(
  -1.1, -0.5,  0.4,
  -1.5, -1.0, -0.2,
   0.3, -0.2, -0.9,
  -0.9,  1.5, -0.6,
   1.8, -1.9,  0.0
), nrow = 5, ncol = 3, byrow = TRUE)

X <- Part2b(n = n_rows, b = b_shift, A = A_mix)

cat("n         =", n_rows,           "\n")
cat("d         =", length(b_shift),  "\n")
cat("k         =", ncol(A_mix),      "\n")
cat("dim(X)    =", dim(X),           "\n")
cat("b:\n");           print(b_shift)
cat("A:\n");           print(A_mix)
cat("AAt (theoretical Cov of each row):\n"); print(A_mix %*% t(A_mix))

n         = 1e+06 
d         = 5 
k         = 3 
dim(X)    = 1000000 5 
b:
[1]  0.67 -0.43 -0.17  1.13  0.29
A:
     [,1] [,2] [,3]
[1,] -1.1 -0.5  0.4
[2,] -1.5 -1.0 -0.2
[3,]  0.3 -0.2 -0.9
[4,] -0.9  1.5 -0.6
[5,]  1.8 -1.9  0.0
AAt (theoretical Cov of each row):
              [,1]  [,2]  [,3]          [,4]  [,5]
[1,]  1.620000e+00  2.07 -0.59  1.110223e-16 -1.03
[2,]  2.070000e+00  3.29 -0.07 -3.000000e-02 -0.80
[3,] -5.900000e-01 -0.07  0.94 -3.000000e-02  0.92
[4,]  1.110223e-16 -0.03 -0.03  3.420000e+00 -4.47
[5,] -1.030000e+00 -0.80  0.92 -4.470000e+00  6.85


## Computing the Results

In [6]:
colmean_first <- colMeans(X)[1]
var_X         <- var(X)
var_1_1       <- var_X[1, 1]
var_1_2       <- var_X[1, 2]

cat("a. colMeans(X)[1] =", colmean_first, "\n")
cat("b. var(X)[1,1]    =", var_1_1,       "\n")
cat("c. var(X)[1,2]    =", var_1_2,       "\n")

a. colMeans(X)[1] = 0.66972 
b. var(X)[1,1]    = 1.626909 
c. var(X)[1,2]    = 2.078758 


## Interpretation

For each row $i$, $\vec{X}_i = A\vec{Z}_i + \vec{b}$ with $\vec{Z}_i \sim \mathcal{N}_k(\vec{0}, I_k)$. Hence:

$$\mathbb{E}[\vec{X}_i] = \vec{b}, \qquad \operatorname{Cov}(\vec{X}_i) = A A^\top$$

- **`colMeans(X)[1]`** estimates $b_1 = 0.67$.
- **`var(X)[1,1]`** estimates $(AA^\top)_{1,1} = \sum_{j=1}^{k} A_{1j}^2 = (-1.1)^2 + (-0.5)^2 + 0.4^2 = 1.21 + 0.25 + 0.16 = 1.62$.
- **`var(X)[1,2]`** estimates $(AA^\top)_{1,2} = \sum_{j=1}^{k} A_{1j} A_{2j} = (-1.1)(-1.5) + (-0.5)(-1.0) + 0.4(-0.2) = 1.65 + 0.5 - 0.08 = 2.07$.

**Implementation note:** we draw a $k \times n$ matrix `Z` of i.i.d. standard Normals, compute $A Z$ (which is $d \times n$), add $\vec{b}$ column-wise via R's recycling on `as.vector(b)`, then transpose to get the $n \times d$ data matrix. `MASS::mvrnorm` is intentionally avoided.

## Final Answer

Build $X = $ `Part2b(n = 10^6, b, A)` with the given $\vec{b}$ and $A$.

With `set.seed(1)`:

$$\text{a.}\;\; \operatorname{colMeans}(X)[1] \approx 0.6697 \qquad \text{b.}\;\; \operatorname{var}(X)[1,1] \approx 1.6269 \qquad \text{c.}\;\; \operatorname{var}(X)[1,2] \approx 2.0788$$

matching the theoretical values $b_1 = 0.67$, $(AA^\top)_{1,1} = 1.62$, $(AA^\top)_{1,2} = 2.07$.